File: src/main/java/com/nugurang/graphql/query/ArticleQuery.java
package com.nugurang.graphql.query;

import com.nugurang.dto.ArticleDto;
import com.nugurang.service.ArticleService;
import graphql.kickstart.tools.GraphQLQueryResolver;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class ArticleQuery implements GraphQLQueryResolver {
    private final ArticleService articleService;

    public Optional<ArticleDto> getArticle(Long id) {
        return articleService.getArticle(id).map(entity -> entity.toDto());
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public ArticleQuery(final ArticleService articleService) {
        this.articleService = articleService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/query/ArticleQuery.kt
package com.nugurang.graphql.query

import com.nugurang.dto.ArticleDto
import com.nugurang.mapper.ArticleMapper
import com.nugurang.service.ArticleService
import graphql.kickstart.tools.GraphQLQueryResolver
import org.springframework.stereotype.Service

@Service
class ArticleQuery(
    private val articleService: ArticleService,
    private val articleMapper: ArticleMapper
) : GraphQLQueryResolver {
    fun getArticle(id: Long): ArticleDto {
        return articleMapper.toDto(articleService.getArticle(id))
    }
}